eng/Versions.props

```diff
@@ -92,7 +92,7 @@
     <Tooling_MicrosoftCodeAnalysisTestingVersion>1.0.1-beta1.21103.2</Tooling_MicrosoftCodeAnalysisTestingVersion>
     <MicrosoftVisualStudioShellPackagesVersion>17.0.31723.112</MicrosoftVisualStudioShellPackagesVersion>
     <MicrosoftVisualStudioPackagesVersion>17.0.487</MicrosoftVisualStudioPackagesVersion>
-    <RoslynPackageVersion>4.1.0-1.21471.13</RoslynPackageVersion>
+    <RoslynPackageVersion>4.2.0-2.22106.6</RoslynPackageVersion>
```

Comment:
💭 I wasn't expecting any changes in this file

Reply:
Without this, we get the same RpcContracts issue that Gen fixed in https://github.com/dotnet/roslyn/pull/59284

I could have referenced RpcContracts directly I suppose. Is there a downside to using a newer roslyn? I'm looking forward to using `!!` in particular :)

Reply:
Downside is third party contributors have no way to run tests locally now. There is no 17.2.